lib/autoprefixer.js

```diff
@@ -31,12 +38,12 @@ module.exports = function () {
         var ifNoneMatch = req.headers['if-none-match'];
         if (ifNoneMatch) {
             var validIfNoneMatchTokens = ifNoneMatch.split(" ").filter(function (etag) {
-                return (/-autoprefixer["-]$/).test(etag);
+                return (/-autoprefixer\[[a-f0-9]{32}\]["-]$/).test(etag);
```

Comment:
I think you'll have to filter by the specific `autoprefixerHash` that you've computed here. Otherwise the "downstream" middleware might return 304 despite the autoprefixer config being different, and you won't get a second chance to process the response body.